src/main.rs

```diff
@@ -1,3 +1,829 @@
+    }
+    for package in &cli.workspace.package {
+        other_options.push("--package".to_string());
+        other_options.push(package.to_string());
+    }
+    for package in &cli.workspace.exclude {
+        other_options.push("--exclude".to_string());
+        other_options.push(package.to_string());
+    }
+    cmd.other_options(other_options);
+
+    info!("Running: {:#?}", cmd.cargo_command());
+
+    let metadata = match cmd.exec() {
+        Ok(metadata) => metadata,
+        Err(e) => {
+            error!("'cargo metadata' failed: {}", e);
+            std::process::exit(-1);
+        }
+    };
+
+    trace!("Got Metadata! {:#?}", metadata);
+
+    //////////////////////////////////////////////////////
+    // Parse out our own configuration
+    //////////////////////////////////////////////////////
+
+    let default_config = Config {
+        version: Some(1),
+        store: Some(Store {
+            path: Some(
+                metadata
+                    .workspace_root
+                    .join(DEFAULT_STORE)
+                    .into_std_path_buf(),
+            ),
+        }),
+        audits: None,
+    };
+
+    let workspace_config = || -> Option<Config> {
+        // FIXME: what is `store.path` relative to here?
+        Config::deserialize(metadata.workspace_metadata.get(WORKSPACE_VET_CONFIG)?)
+            .map_err(|e| {
+                error!(
+                    "Workspace had [{WORKSPACE_VET_CONFIG}] but it was malformed: {}",
+                    e
+                );
+                std::process::exit(-1);
+            })
+            .ok()
+    }();
+
+    let package_config = || -> Option<Config> {
+        // FIXME: what is `store.path` relative to here?
+        Config::deserialize(metadata.root_package()?.metadata.get(PACKAGE_VET_CONFIG)?)
+            .map_err(|e| {
+                error!(
+                    "Root package had [{PACKAGE_VET_CONFIG}] but it was malformed: {}",
+                    e
+                );
+                std::process::exit(-1);
+            })
+            .ok()
+    }();
+
+    let mut configs = vec![default_config];
+    if let Some(cfg) = workspace_config {
+        configs.push(cfg);
+    }
+    if let Some(cfg) = package_config {
+        configs.push(cfg);
+    }
+    let config = Configs(configs);
+
+    info!("Final Config: ");
+    info!("  - version: {}", config.version());
+    info!("  - store.path: {:#?}", config.store_path());
+    info!("  - audits: {:#?}", config.audits());
+
+    //////////////////////////////////////////////////////
+    // Run the actual command
+    //////////////////////////////////////////////////////
+
+    let init = is_init(&config);
+    if matches!(cli.command, Some(Commands::Init { .. })) {
+        if init {
+            error!(
+                "'cargo vet' already initialized (store found at {:#?})",
+                config.store_path()
+            );
+            std::process::exit(-1);
+        }
```

Comment:
Should re-initing a repository do something useful?

Reply:
Probably just throwing for now.